app/src/main/java/com/matmoongi/viewmodels/MainViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package com.matmoongi.viewmodels
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import com.matmoongi.data.FavoritesRepository
+import com.matmoongi.data.MainRestaurant
+import com.matmoongi.data.RestaurantsRepository
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+class MainViewModel(
+    private val favoritesRepository: FavoritesRepository,
+    private val restaurantsRepository: RestaurantsRepository,
+    private val state: SavedStateHandle,
+) : ViewModel() {
+
+    private val restaurantsState: StateFlow<List<MainRestaurant>> = state.getStateFlow(
+        "restaurantsState",
+        emptyList(),
+    )
+
+    init {
+        // 앱 시작 시 현 위치로 음식점 검색
+        CoroutineScope(Dispatchers.IO).launch { state["restaurantsState"] = onClickSearchButton() }
+    }
+
+    /** 주변 음식점을 검색해서 받아오기 - 현재 위치 기반 **/
+    private suspend fun onClickSearchButton(): List<MainRestaurant> {
+        restaurantsRepository.getNearbyRestaurants()
+        TODO("MainRestaurant로 가공해서 리턴")
+    }
```

Comment:
메소드들 간의 호응이 뭔가 어울리지 않는 것 같습니다.

1. 이벤트 발생 상황이 아닌데, `onClickSearchButton()`이 호출됨
2. 이벤트 발생 콜백 함수인 `onClickSearchButton()`이 반환값을 가지고 있음

```suggestion
    init {
        // 앱 시작 시 현 위치로 음식점 검색
        coroutineScope.launch { pullNearbyRestaurants() }
    }

    /** 주변 음식점을 검색해서 받아오기 - 현재 위치 기반 **/
    private fun onClickSearchButton() {
        coroutineScope.launch { pullNearbyRestaurants() }
    }
    
    private suspend fun refreshNearbyRestaurantList() {
      val restaurant = withContext(Dispatchers.IO) { restaurantsRepository.getNearbyRestaurants() }
      state["restaurantState"] = restaurant.map { it.toMainRestaurant() }
    }
    
    private fun Restaurant.toMainRestaurant(): MainRestaurant() =
        TODO("...")
```